languagetool-language-modules/ar/src/main/java/org/languagetool/rules/ar/AdjectiveToExclamationFilter.java

```diff
@@ -0,0 +1,204 @@
+/* LanguageTool, a natural language style checker
+ * Copyright (C) 2022 Sohaib Afifi, Taha Zerrouki
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+package org.languagetool.rules.ar;
+
+import org.jetbrains.annotations.Nullable;
+import org.languagetool.AnalyzedTokenReadings;
+import org.languagetool.rules.RuleMatch;
+import org.languagetool.rules.SimpleReplaceDataLoader;
+import org.languagetool.rules.patterns.RuleFilter;
+import org.languagetool.tagging.ar.ArabicTagger;
+import org.languagetool.tools.ArabicWordMaps;
+
+import java.util.*;
+
+/**
+ * Filter that maps suggestion from adverb to adjective.
+ *
+ * @since 5.8
+ */
+public class AdjectiveToExclamationFilter extends RuleFilter {
+
+  public AdjectiveToExclamationFilter() {
+    this.adj2compList = loadFromPath(FILE_NAME);
+  }
+
+  private final ArabicTagger tagger = new ArabicTagger();
+  private static final String FILE_NAME = "/ar/arabic_adjective_exclamation.txt";
+  private final Map<String, List<String>> adj2compList;
+
+  private final Map<String, String> adj2comp = new HashMap<String, String>() {{
+    // tri letters verb:
+    put("رشيد", "أرشد");
+    put("طويل", "أطول");
+    put("بديع", "أبدع");
+    //
+    // TODO: add more Masdar verb
+    //put("", "");
+  }};
+
+
+  @Nullable
+  @Override
+  public RuleMatch acceptRuleMatch(RuleMatch match, Map<String, String> arguments, int patternTokenPos, AnalyzedTokenReadings[] patternTokens) {
+
+    //  This rule return only the comparative according to given adjective
+
+    String adj = arguments.get("adj"); // extract adjective
+    String noun = arguments.get("noun"); // the second argument
+    int adjTokenIndex;
+    try {
+      adjTokenIndex = Integer.valueOf(arguments.get("adj_pos")) - 1;
+    } catch (NumberFormatException e) {
+      e.printStackTrace();
```

Comment:
We shouldn't print stacktraces like this - if this is an internal bug, the exception should be thrown. Otherwise, it should be logged with the logging framework we use elsewhere.